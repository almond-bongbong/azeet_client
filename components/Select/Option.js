import React from 'react';
import PropTypes from 'prop-types';

const Option = ({ text, value }) => (
  <option value={value}>{text}</option>
);

Option.propTypes = {
  text: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default Option;
