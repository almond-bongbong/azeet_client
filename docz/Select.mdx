---
name: Select
---

import { Playground, Props } from 'docz'
import Select from 'components/Select/Select'
import Option from 'components/Select/Option'

# Select

<Props of={Select} />

## Basic usage

<Playground>
  {() => {
    const Test = () => {
      const [value, setValue] = React.useState(0);

      return (
        <Select value={value} onChange={e => setValue(e.target.value)}>
          <Option text="서울" value={0} />
          <Option text="경기도" value={1} />
          <Option text="부산" value={2} />
          <Option text="인천" value={3} />
          <Option text="제주도" value={4} />
        </Select>
      )
    }
    return (
      <Test />
    )
  }}
</Playground>

## Theme

### line

<Playground>
  <Select theme="line" value={0} onChange={e => console.log(e)}>
    <Option text="서울" value={0} />
    <Option text="경기도" value={1} />
    <Option text="부산" value={2} />
    <Option text="인천" value={3} />
    <Option text="제주도" value={4} />
  </Select>
</Playground>

## Inline

<Playground>
  <Select inline={true} value={0} onChange={e => console.log(e)}>
    <Option text="서울" value={0} />
    <Option text="경기도" value={1} />
    <Option text="부산" value={2} />
    <Option text="인천" value={3} />
    <Option text="제주도" value={4} />
  </Select>
</Playground>

## Size

### 30

<Playground>
  <Select size={30} value={0} onChange={e => console.log(e)}>
    <Option text="서울" value={0} />
    <Option text="경기도" value={1} />
    <Option text="부산" value={2} />
    <Option text="인천" value={3} />
    <Option text="제주도" value={4} />
  </Select>
</Playground>

### 40 (default)

<Playground>
  <Select size={40} value={0} onChange={e => console.log(e)}>
    <Option text="서울" value={0} />
    <Option text="경기도" value={1} />
    <Option text="부산" value={2} />
    <Option text="인천" value={3} />
    <Option text="제주도" value={4} />
  </Select>
</Playground>

### 50

<Playground>
  <Select size={50} value={0} onChange={e => console.log(e)}>
    <Option text="서울" value={0} />
    <Option text="경기도" value={1} />
    <Option text="부산" value={2} />
    <Option text="인천" value={3} />
    <Option text="제주도" value={4} />
  </Select>
</Playground>
