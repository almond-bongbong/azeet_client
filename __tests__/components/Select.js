import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Select from 'components/Select/Select';
import Option from 'components/Select/Option';

describe('Select', () => {
  it('matches snapshot', () => {
    const utils = render(
      <Select onChange={_ => _} value="0">
        <Option text="첫번째" value="0" />
        <Option text="두번째" value="1" />
        <Option text="세번째" value="2" />
      </Select>,
    );

    expect(utils.container).toMatchSnapshot();
  });

  it('selected second item', () => {
    const { getByDisplayValue, queryByDisplayValue } = render(
      <Select onChange={_ => _} value="1">
        <Option text="첫번째" value="0" />
        <Option text="두번째" value="1" />
        <Option text="세번째" value="2" />
      </Select>,
    );

    getByDisplayValue('두번째');
    expect(queryByDisplayValue('첫번째')).toBeNull();
  });

  it('calls onChange', () => {
    const handleChange = jest.fn();
    const { getByDisplayValue } = render(
      <Select onChange={handleChange} value="0">
        <Option text="첫번째" value="0" />
        <Option text="두번째" value="1" />
        <Option text="세번째" value="2" />
      </Select>,
    );

    const select = getByDisplayValue('첫번째');
    fireEvent.change(select, { target: { value: '1' } });
    expect(handleChange).toBeCalled();
  });
});
