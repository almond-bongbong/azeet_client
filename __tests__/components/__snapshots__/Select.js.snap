// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select matches snapshot 1`] = `
<div>
  <div
    class="Select__BasicSelectStyle-sc-1x647ba-0 iqzeTB"
    size="40"
  >
    <select>
      <option
        value="0"
      >
        첫번째
      </option>
      <option
        value="1"
      >
        두번째
      </option>
      <option
        value="2"
      >
        세번째
      </option>
    </select>
  </div>
</div>
`;
